From d684a1f21883c8ffc43c9aff8b1d2bb526e15cc3 Mon Sep 17 00:00:00 2001
From: Mikael Magnusson <mikma@users.sourceforge.net>
Date: Mon, 2 Sep 2019 14:53:20 +0200
Subject: [PATCH] state: support reboot of containers

---
 state.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/state.c b/state.c
index 4737d01..0ca3dca 100644
--- a/state.c
+++ b/state.c
@@ -172,8 +172,10 @@ static void state_enter(void)
 
 			while (1)
 				sleep(1);
-		} else
+		} else {
+			reboot(reboot_event);
 			exit(0);
+		}
 #else
 		exit(0);
 #endif
-- 
2.17.1

